modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ClientImpl.java

```diff
@@ -213,6 +214,8 @@
      */
     ClientImpl(TcpDiscoverySpi adapter) {
         super(adapter);
+        executorService = Executors.newSingleThreadScheduledExecutor(
+            new IgniteThreadFactory("client-node", "tcp-discovery-exec"));
```

Comment:
client-node -> adapter.ignite().name(); (null check should be added for adapter.ignite())